drivers/sensor/bmi270/bmi270.c

```diff
@@ -715,33 +773,69 @@ static int bmi270_init(const struct device *dev)
 static const struct sensor_driver_api bmi270_driver_api = {
 	.sample_fetch = bmi270_sample_fetch,
 	.channel_get = bmi270_channel_get,
-	.attr_set = bmi270_attr_set
+	.attr_set = bmi270_attr_set,
+#if defined(CONFIG_BMI270_TRIGGER)
+	.trigger_set = bmi270_trigger_set,
+#endif
 };
 
+static const struct bmi270_feature_config bmi270_feature_max_fifo = {
+	.name = "max_fifo",
+	.config_file = bmi270_config_file_max_fifo,
+	.config_file_len = sizeof(bmi270_config_file_max_fifo),
+};
+
+static const struct bmi270_feature_config bmi270_feature_base = {
+	.name = "base",
+	.config_file = bmi270_config_file_base,
+	.config_file_len = sizeof(bmi270_config_file_base),
+	.anymo_1 = &(struct bmi270_feature_reg){ .page = 1, .addr = 0x3C },
+	.anymo_2 = &(struct bmi270_feature_reg){ .page = 1, .addr = 0x3E },
+};
+
+static const struct bmi270_feature_config bmi270_feature_legacy = {
+	.name = "legacy",
+	.config_file = bmi270_config_file_legacy,
+	.config_file_len = sizeof(bmi270_config_file_legacy),
+};
```

Comment:
What is the difference between legacy and base? Can you please explain the changes to the config blob?

Reply:
The BMI270 is basically firmware flashed at each boot. The blobs enable entirely different feature sets. To get triggers, you can't use max_fifo. This driver doesn't implement any features from the `legacy` blob tho so I guess that part could be removed...

Reply:
But the new firmware blob breaks compatibility. It's essentially becomes a different device, albeit using the same silicon. To me this seems like a pretty good place to utilize the compat string. I think making a whole new driver would be more close to correct than introducing a property.